Algorithm/QCAlgorithm.Indicators.cs

```diff
@@ -3149,7 +3151,10 @@ private IEnumerable<Slice> GetIndicatorWarmUpHistory(IEnumerable<Symbol> symbols
 
                 try
                 {
-                    return History(symbols, periods, resolution, dataNormalizationMode: GetIndicatorHistoryDataNormalizationMode(indicator));
+                    var symbols = new[] { symbol };
+                    CheckPeriodBasedHistoryRequestResolution(symbols, resolution, null);
+                    historyRequest = CreateBarCountHistoryRequests(symbols, periods, resolution, dataNormalizationMode: GetIndicatorHistoryDataNormalizationMode(indicator)).Single();
+                    return GetSlicesFromHistoryRequests(historyRequest);
```

Comment:
Using the History() API might be enough, if you only need the request end time you could use QCAlgorithm.Time or UtcTime and convert it to the security's time zone to do the final consolidator scan